### PR TITLE
Add limit token filter docs

### PR DESCRIPTION
**Pull request by AntonEliatra**
### Description
Add limit token filter docs

### Issues Resolved
Closes https://github.com/opensearch-project/documentation-website/issues/8153

### Version
all

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by vagimeli** [comment]
@udabhas Will you review this PR for technical accuracy? Thank you.

**Comment by vagimeli** [comment]
> @udabhas Will you review this PR for technical accuracy? Thank you.

@udabhas @varun-lodaya This is a month old. We need tech review approval to move it forward in the documentation process. Please review this week or provide a peer who can review it. Thank you.

File: _analyzers/token-filters/index.md [hunk 1]
```diff
@@ -37,7 +37,7 @@ Token filter | Underlying Lucene token filter|  Description
...
 `length` | [LengthFilter](https://lucene.apache.org/core/9_10_0/analysis/common/org/apache/lucene/analysis/miscellaneous/LengthFilter.html) | Removes tokens whose lengths are shorter or longer than the length range specified by `min` and `max`. 
-`limit` | [LimitTokenCountFilter](https://lucene.apache.org/core/9_10_0/analysis/common/org/apache/lucene/analysis/miscellaneous/LimitTokenCountFilter.html) | Limits the number of output tokens. A common use case is to limit the size of document field values based on token count.
+[`limit`]({{site.url}}{{site.baseurl}}/analyzers/token-filters/limit/) | [LimitTokenCountFilter](https://lucene.apache.org/core/9_10_0/analysis/common/org/apache/lucene/analysis/miscellaneous/LimitTokenCountFilter.html) | Limits the number of output tokens. A common use case is to limit the size of document field values based on token count.
```
Comment (natebower) [inline _analyzers/token-filters/index.md:40]:
```suggestion
[`limit`]({{site.url}}{{site.baseurl}}/analyzers/token-filters/limit/) | [LimitTokenCountFilter](https://lucene.apache.org/core/9_10_0/analysis/common/org/apache/lucene/analysis/miscellaneous/LimitTokenCountFilter.html) | Limits the number of output tokens. For example, document field value sizes can be limited based on the token count.
```

File: _analyzers/token-filters/limit.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
+---
+layout: default
+title: Limit
+parent: Token filters
+nav_order: 250
+---
+
+# Limit token filter
+
+The `limit` token filter is used to limit the number of tokens that are passed through the analysis chain.
```
Comment (natebower) [inline _analyzers/token-filters/limit.md:10]:
```suggestion
The `limit` token filter is used to limit the number of tokens passed through the analysis chain.
```

File: _analyzers/token-filters/limit.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+## Parameters
+
+The `limit` token filter can be configured with the following parameters.
+
+Parameter | Required/Optional | Data type | Description
+:--- | :--- | :--- | :--- 
+`max_token_count` | Optional | Integer | The maximum number of tokens to be generated. Default is `1`.
+`consume_all_tokens` | Optional | Boolean | (Expect level setting) Use all tokens from tokenizer, even if result exceeds `max_token_count`. When this parameter is set, the output still only contains the number of tokens specified in `max_token_count`. However, all tokens generated by the tokenizer are processed. Default is `false`.
```
Comment (natebower) [inline _analyzers/token-filters/limit.md:19]:
```suggestion
`consume_all_tokens` | Optional | Boolean | (Expect level setting) Uses all tokens from the tokenizer, even if the result exceeds `max_token_count`. When this parameter is set, the output still only contains the number of tokens specified by `max_token_count`. However, all tokens generated by the tokenizer are processed. Default is `false`.
```

File: _analyzers/token-filters/limit.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
```
Comment (natebower) [inline _analyzers/token-filters/limit.md:20]: Line 19: What does "(Expect level setting)" mean?

**Review by natebower — COMMENTED** [review]
@kolchfa-aws One comment and a few changes. Thanks!

File: _analyzers/token-filters/limit.md (hunk 4 above)
Comment (kolchfa-aws) [reply _analyzers/token-filters/limit.md:20]: Expert 😄 Thanks for the catch.

Comment (natebower) [reply _analyzers/token-filters/limit.md:20]: YW, but should be "Expert-level setting" in that case 😄

**Review by kolchfa-aws — APPROVED** [review]

**Pull request merged**